### DISCUSSION TITLE
mise use ubi:helix-editor/helix

### DISCUSSION THREAD

**Discussion by cysabi** [jdx/mise#2916] [Q&A]
hi! when installing helix using the `ubi` backend, you are unable to run `hx`.

- helix does work with an asdf community plugin
- i have tried setting a `{ exe: "hx" }` to no avail
- using `mise bin-paths` reveals that all the bin folders for `/root/.local/share/mise/installs/ubi-helix-editor-helix` is empty

**Comment by kemitix** [comment]
I can get helix to install *from source* with:

```toml
"cargo:https://github.com/helix-editor/helix" = { version = "tag:25.01", bin = "hx", crate = "helix-term", locked = "true" }
```

I'd prefer to use use ubi, but the binary release from helix isn't compatible with Debian 12 (glibc mismatch).

In either case, there is still the manual step of pointing HELIX_RUNTIME to the `runtime` folder within the source tree.

**Comment by Marukome0743** [comment]
A quick update for anyone finding this discussion: Helix is now available through mise’s registry using the Aqua backend, so the original `ubi:` installation issue no longer applies.

I verified this on the current main branch with:

```sh
mise x helix@latest -- hx --version
```

This installed Helix 25.07.1 successfully and produced:

```text
helix 25.07.1 (a05c151b)
```

The bundled runtime was also detected automatically from the mise installation directory when running `hx --health`, so setting `HELIX_RUNTIME` manually was not necessary.

The normal installation can now be written as:

```sh
mise use helix@latest
```

or, using the backend explicitly:

```sh
mise use aqua:helix-editor/helix@latest
```

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
